### PR TITLE
xhtml2pdf report template

### PR DESCRIPTION
**Pull request by rutujaac**
Non Pango Report PDF Template using xhtml2pdf:

1. Introduces new function to generate PDF report from HTML Template using xhtml2pdf python library

_Pass the value to **renderer** in **Reports.generate_report** function in reports.py_
_Default value for renderer is **weasyprint**_

```

Reports.generate_report(data, outputPath='./report.pdf', format = 'pdf', renderer='xhtml2pdf')


```

Scenarios tested:
1. Print report using default `weasyprint` renderer:
Output: 
image

2. Print report using `xhtml2pdf` renderer:
Output:
image

3. Unsupported output file `format`:
Ouput: 
image

4. Unsupported renderer:
Output: 
image

### CODE REVIEW

File: pebblo/reports/reports.py [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
+    def generate_report(data, outputPath='./report.pdf', format = 'pdf', renderer = ReportLibraries.WEASYPRINT):
+        if format == 'pdf':
+            searchPath = os.path.join(os.path.dirname(__file__), 'templates/')
+            try:
+                templateName = template_renderer_mapping[renderer]
+                convertHtmlToPdf(data, outputPath, templateName = templateName, searchPath = searchPath, renderer = renderer)
+            except Exception as e:
+                logger.error(f"Renderer {renderer} not supported. Please use supported renderers: {ReportLibraries.WEASYPRINT} or {ReportLibraries.XHTML2PDF}, {e}")
+        else:
+            logger.error("Output file format not supported")
```
Comment (shreyas-damle) [inline pebblo/reports/reports.py:20]: nit pick - please add format name in the error message, so that end user can see what value was passed.

**Review by shreyas-damle — CHANGES_REQUESTED** [review]
- Please add output with 
    - Supported format and renderer
    - Supported format and unsupported renderer
    - Unsupported format and supported renderer
- I would suggest to add unit tests with this, so that we make sure that existing functionality is not breaking.

File: pebblo/reports/reports.py (hunk 1 above)
Comment (rutujaac) [reply pebblo/reports/reports.py]: Added format string

**Review by shreyas-damle — DISMISSED** [review]
Changes look good to me. Please consider taking unit tests in next PR.

**Review by shreyas-damle — APPROVED** [review]

**Pull request merged**